### DISCUSSION TITLE
Does send_recv.c test missing a call for io_uring_queue_exit()?

### DISCUSSION THREAD

**Discussion by itaysnir** [axboe/liburing#718] [Q&A] [answered]
Hey everyone,

Im pretty new here, sorry if im asking this in the wrong place.
Lately I've browsed: https://github.com/axboe/liburing/blob/master/test/send_recv.c .

I've seen the function do_send() allocates a ring via io_uring_queue_init() call.
However, it doesn't tears down the ring via io_uring_queue_exit(). 

Its not a big issue (doesn't impact the test correctness by any means) but i'd still like to fix this incase its wrong.

Please let me know if this is wrong, and ill open a PR. 

Thank you.

**Comment by ammarfaizi2** [comment, marked as answer]
Yes, we are leaking a ring in do_send(). While it is not a serious issue as it is just a test program, I think we should start being serious in writing them, especially people will rely on the those tests when looking for examples.

> Please let me know if this is wrong, and ill open a PR.

Yes, it is wrong. Go ahead create a PR.

**Reply by itaysnir** [reply]
Thank you for your quick answer, will do :)
